consensus/ibft/src/main/java/tech/pegasys/pantheon/consensus/ibft/statemachine/IbftBlockHeightManagerFactory.java

```diff
@@ -31,15 +31,20 @@ public IbftBlockHeightManagerFactory(
     this.messageValidatorFactory = messageValidatorFactory;
   }
 
-  public IbftBlockHeightManager create(final BlockHeader parentHeader) {
-    return new IbftBlockHeightManager(
-        parentHeader,
-        finalState,
-        new RoundChangeManager(
-            IbftHelpers.calculateRequiredValidatorQuorum(finalState.getValidators().size()),
-            messageValidatorFactory.createRoundChangeMessageValidator(parentHeader)),
-        roundFactory,
-        finalState.getClock(),
-        messageValidatorFactory);
+  public BlockHeightManager create(final BlockHeader parentHeader) {
+
+    if (finalState.isLocalNodeValidator()) {
+      return new IbftBlockHeightManager(
+          parentHeader,
+          finalState,
+          new RoundChangeManager(
+              IbftHelpers.calculateRequiredValidatorQuorum(finalState.getValidators().size()),
+              messageValidatorFactory.createRoundChangeMessageValidator(parentHeader)),
+          roundFactory,
+          finalState.getClock(),
+          messageValidatorFactory);
+    } else {
+      return new NoOpBlockHeightManager(parentHeader);
```

Comment:
think it would be better to have a seperate method for creating the NoOpBlockHeightManager then all the logic for non-validator nodes can be moved/stay in the IbftController